grpc/shared_conn.go

```diff
@@ -85,6 +88,42 @@ type SharedConn struct {
 	logger logging.Logger
 }
 
+// NewSharedConnForModule acts as a constructor for `SharedConn` for modules that are communicating
+// back to their parent viam-server.
+func NewSharedConnForModule(grpcConn rpc.ClientConn, peerConn *webrtc.PeerConnection, logger logging.Logger) *SharedConn {
+	// We must be passed a ready connection.
+	pcReady := make(chan struct{})
+	close(pcReady)
+
+	ret := &SharedConn{
+		peerConn:      peerConn,
+		peerConnReady: pcReady,
+		// We were passed in a ready connection. Only create this for when `Close` is called.
+		peerConnFailed:       make(chan struct{}),
+		onTrackCBByTrackName: make(map[string]OnTrackCB),
+		logger:               logger,
+	}
+	ret.grpcConn.ReplaceConn(grpcConn)
+
+	ret.peerConn.OnTrack(func(trackRemote *webrtc.TrackRemote, rtpReceiver *webrtc.RTPReceiver) {
+		sid := trackRemote.StreamID()
+		if sid == "rtsp-1" {
```

Comment:
@nicksanford , we need to do a real thing here. Not sure if the logic here is simple enough to communicate over a PR or if syncing in person would yield a better answer.

Reply:
lets sync in person about this

Reply:
This was my hack solution: 
https://github.com/viamrobotics/rdk/pull/4737/files#diff-f789e424ac8cea76b93d85718d17db1578e00d0c5de523e5267df82bc8111381R103-R129